### ISSUE TITLE
Wrong asd value?

### ISSUE THREAD

**Issue by frederic-michaud** [sapfo/medeas#22]
(Not sure this is my mistake or something wrong in the code, but I have no time to look at this now so I post not to forget about it)

I'm simulating population with 10'000 loci, but when I looked at the distance matrix (asd files), it seems that all number are multiple of 9'900. Given formula 5 of the draft, I don't see how that is possible. They might be some mistake in the way asd is computed, like a loop not going until the end.

**Comment by ilevkivskyi** [comment]
Also just a guess, but isn't this the result of statistical bootstrap? The asd matrix is computed many times (you should see many asd files). There is only one file with all loci taken into account, in all other we remove some random blocks to get "statistics" of inferred `D`, `D'` etc.

**Comment by frederic-michaud** [comment]
I think that the problem is the same in all bootstraped sample (and the not bootstraped sample):
```
f = open("/Users/fmichaud/PycharmProjects/medeas/test/pp.1.asd.data.boot.1","rb")
mat_asd = pickle.load(f)
mat_asd[0:5,0:5]*9900
array([[  0., 530., 523., 491., 495.],
       [530.,   0., 505., 495., 499.],
       [523., 505.,   0., 494., 508.],
       [491., 495., 494.,   0., 468.],
       [495., 499., 508., 468.,   0.]])
f = open("/Users/fmichaud/PycharmProjects/medeas/test/pp.1.asd.data.boot.2","rb")
mat_asd = pickle.load(f)
mat_asd[0:5,0:5]*9900
array([[  0., 589., 558., 548., 502.],
       [589.,   0., 541., 501., 475.],
       [558., 541.,   0., 514., 504.],
       [548., 501., 514.,   0., 476.],
       [502., 475., 504., 476.,   0.]])
```
Actually, as far as I understand how you bootstrap, you always consider the same number of loci:
```
clen = len(chunk_data)  
for i in range(clen):
    chunk_res[i] = chunk_data[randint(0, clen - 1)]
```

**Comment by frederic-michaud** [comment]
I'm still not sure about the exact logic of make_asd.py, but is it possible that you let some data here: https://github.com/sapfo/medeas/blob/master/src/make_asd.py#L163 that are never taken into account?

I mean, for a very large file, you will call several time process_chunks, and the remainder will pass from one to another call I guess. But when you reach the end of the file, what is in remainder will stay in remainder without being taken into account in tot_dists and tot_norms. 
This is not really important as long as the amount of data is bigger than BOOTSIZE, which should always be the case if we chose correctly BOOTSIZE.

(My boostrap windows is 100, and I have actually about 9995 loci and not 10'000 since sometime no mutation appears on the coalescence tree, 10'000 - 100 = 9900).

**Comment by ilevkivskyi** [comment]
The `remainder` variable is for attaching the part that is not divisible by `BOOTSIZE` to the next chromosome. Imagine you have 250 sites on first chromosome and `BOOTSIZE = 100`. The there will be two chunks on first chromosome, and the last 50 will be combined with first 50 in the second chromosome. Then if the second chromosome has 220 sites, then we get: 50 from first plus 50 from second, then 100 from second, then remaining 70 will be combined with first 30 from the third chromosome and so on.

The idea is we have one single very long chromosome stitched from all 23, and then we cut in in chunks of 100 sites.

Btw, now that explain this it seems to me that the last remainder on 23rd chromosome (smaller than 100, like 95 sites in your case) is just thrown away. This should be then easy to fix.

**Comment by frederic-michaud** [comment]
Yes yes, this is what I meant (except I consider the loop https://github.com/sapfo/medeas/blob/master/src/make_asd.py#L172 for simulated data and you consider the loop https://github.com/sapfo/medeas/blob/master/src/make_asd.py#L180 for real data). The last part is thrown away. And this is why I have only 9900 loci instead of 9995. 

Yes, would be easy to fix, though I'm not 100% sure that I want to fix this. Then different bootstraped matrices might come from different number of loci. Not sure that the bias that it would introduce is better or worst than biais for throwing away some data. Idealy we should adapt the BOOTSIZE to the number of loci... I'll think about it.
